int/api/events.go

```diff
@@ -0,0 +1,67 @@
+package api
+
+import (
+	"strings"
+	"time"
+
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/massalabs/thyra/api/swagger/server/models"
+	"github.com/massalabs/thyra/api/swagger/server/restapi/operations"
+	"github.com/massalabs/thyra/pkg/node"
+)
+
+//nolint:nolintlint,ireturn
+func EventListenerHandler(params operations.ThyraEventsGetterParams) middleware.Responder {
+	const periodsBefore = 2
+
+	client := node.NewDefaultClient()
+
+	status, err := node.Status(client)
+	if err != nil {
+		return operations.NewThyraEventsGetterInternalServerError().
+			WithPayload(
+				&models.Error{
+					Code:    errorCodeGetDomainNames,
+					Message: err.Error(),
+				})
+	}
+
+	slotStart := node.Slot{
+		Period: status.LastSlot.Period - periodsBefore,
```

Comment:
Can status.LastSlot.Period be smaller than 2 ?
If that is the case we may end up with Period being negative in some cases. I don't know how it will impact the rest of the code (maybe node.Events will just throw an error) but I'm not sure this is the desired behavior.